docs/operations.rst

```diff
@@ -14,42 +14,169 @@ Provisioning
+	 currently reported and goal state are the same.
+
+	 Run the follow SQL query on the monitor database to enable the
+	 maintenance operation::
+
+		select nodeid, nodename, nodeport,
+		       pgautofailover.start_maintenance(nodename, nodeport)
+		  from pgautofailover.node
+		 where reportedstate = 'secondary'
+		   and not exists
+			  (
+			   select nodeid
+				 from pgautofailover.node
+				where goalstate <> reportedstate
+				   or reportedstate not in ('primary', 'secondary')
+			  )
+		order by nodeid;
+
+	 Check that the selected nodes have reached `maintenance` by watching
+	 the output of the ``pg_autoctl show state`` command. When all selected
+	 nodes are in maintenance, we can move forward with the upgrade.
+
+	 .. note::
+
+		Alternatively you may also run the command ``pg_autoctl enable
+		maintenance`` on every secondary node.
+
+  2. Stop the ``pg_autoctl`` service on the monitor.
+
+	 The command to stop the systemd pgautofailover service is::
+
+	   sudo systemctl stop pgautofailover
+
+	 To verify that the service has been stopped as expected, we run the
+	 following command::
+
+	   sudo systemctl status pgautofailover
+
+	 At this point all the Postgres nodes in the system are going to error
+	 out every second when connecting to the monitor. In case of error to
+	 connect to the monitor, the node-active process retries every second,
+	 and will be able to continue normally as soon as the monitor is back
+	 online.
+
+  3. Upgrade the pg_auto_failover package on the monitor.
+
+	 When using a debian based OS, this looks like the following command for
+	 pg_auto_failover 1.3.1::
+
+	   sudo apt-get remove pg-auto-failover-cli-enterprise-1.0 postgresql-11-auto-failover-enterprise-1.0
+	   sudo apt-get install -q -y pg-auto-failover-cli-enterprise-1.3 postgresql-11-auto-failover-enterprise-1.3
+
+  4. Restart the ``pgautofailover`` service on the monitor.
+
+	 When using the systemd integration, all we need to do is::
+
+	   sudo systemctl start pgautofailover
+
+	 Then we may use the following commands to make sure that the service is
+	 running as expected::
+
+	   sudo systemctl status pgautofailover
+	   sudo journalctl -u pgautofailover
+
+	 At this point it is expected that the ``pg_autoctl`` logs show that an
+	 upgrade has been performed by using the ``ALTER EXTENSION
+	 pgautofailover UPDATE TO ...`` command. The monitor is ready with the
+	 new version of pg_auto_failover.
+
+  5. Upgrade the pg_auto_failover package on the Postgres nodes.
+
+	 Use the same command as on the monitor in step 4 above.
+
+  6. Restart the ``node-service`` service on all the Postgres nodes.
+
+	 When using the systemd integration, restarting the ``pgautofailover``
+	 service will also restart Postgres, which is not always possible. If
+	 you did implement step 2, your current service is not running, so you
+	 can restart it now::
+
+	   pg_autoctl restart node-active
```

Comment:
inconsistent use of `node-service`/`node-active`

Reply:
Fixed.